src/pipewire/pipewire.cc

```diff
@@ -164,31 +178,60 @@ void PipeWireOutput::pause(bool pause)
 
 int PipeWireOutput::get_delay()
 {
-    return (m_buffer_at / m_stride + m_frames) * 1000 / m_rate;
+    int buff_time = ((m_buffer.len() / m_stride) * 1000) / m_rate;
+    int pw_buff_time = ((m_pw_buffer_size / m_stride) * 1000) / m_rate;
+    int time_diff = 0;
+    int add_delay = 0;
+
+    // Get time difference from updated time snapshot of the stream
+    struct pw_time time;
+    if (pw_stream_get_time_n(m_stream, &time, sizeof time) == 0)
+    {
+        time_diff = (pw_stream_get_nsec(m_stream) - time.now) / SPA_NSEC_PER_MSEC;
+        time_diff = aud::clamp(time_diff, 0, pw_buff_time);
+        add_delay = time.queued * 1000 / m_rate;
+
+#if PW_CHECK_VERSION(0, 3, 50)
+        add_delay += time.buffered * 1000 / m_rate;
+#endif
+        if (time.rate.denom > 0)
+            add_delay += time.delay * 1000 * time.rate.num / time.rate.denom;
+    }
+
+    return buff_time + pw_buff_time - time_diff + add_delay;
 }
```

Comment:
@maris-ab: `pw_time.buffered` is only available since 0.3.50.
I have split `add_delay = (v_time.buffered + v_time.queued) * 1000 / m_rate;` into two separate calculations.

Also changed
`time_diff = aud::clamp<int>(0, time_diff, pw_buff_time);`
to
`time_diff = aud::clamp(time_diff, 0, pw_buff_time);`